### PR TITLE
refactor!: Use common units for seed finding

### PR DESCRIPTION
**Pull request by paulgessinger**
The seed finding config currently uses implicit units. This PR switches to explicit units using the common `Acts::UnitConstants`. Since we don't want to do the conversion to seeding-internal units all the time, it does this once in the constructor, where special methods `toInternalUnits` are called which return a copy of the config with the members converted to internal units.

This should result in no changes to the actual calculations, and only to the configuration interface.

BREAKING CHANGE:
The seeding configuration structs `SpacePointGridConfig`, `SeedFilterConfig` and `SeedFinderConfig` change to now use ACTS units.

### CODE REVIEW

File: Core/include/Acts/Seeding/SeedFilterConfig.hpp [hunk 1]
```diff
@@ -16,21 +18,30 @@ namespace Acts {
 struct SeedFilterConfig {
   // the allowed delta between two inverted seed radii for them to be considered
   // compatible.
-  float deltaInvHelixDiameter = 0.00003;
+  float deltaInvHelixDiameter = 0.00003 * 1. / Acts::UnitConstants::mm;
```
Comment (paulgessinger) [inline Core/include/Acts/Seeding/SeedFilterConfig.hpp:21]: @robertlangenberg does this make sense?

File: Core/include/Acts/Seeding/SeedfinderConfig.hpp [hunk 2]
```diff
@@ -44,37 +45,38 @@ struct SeedfinderConfig {
...
   // impact parameter in mm
   // FIXME: Acts units
-  float impactMax = 20.;
+  float impactMax = 20. * Acts::UnitConstants::mm;
 
   // how many sigmas of scattering angle should be considered?
   float sigmaScattering = 5;
   // Upper pt limit for scattering calculation
-  float maxPtScattering = 10000;
+  float maxPtScattering = 10 * Acts::UnitConstants::GeV;
```
Comment (paulgessinger) [inline Core/include/Acts/Seeding/SeedfinderConfig.hpp:53]: @robertlangenberg is this correct?

File: Core/include/Acts/Seeding/SeedfinderConfig.hpp [hunk 3]
```diff
@@ -87,8 +89,8 @@ struct SeedfinderConfig {
   // will be added to spacepoint measurement uncertainties (and therefore also
   // multiplied by sigmaError)
   // FIXME: call align1 and align2
-  float zAlign = 0;
-  float rAlign = 0;
+  float zAlign = 0 * Acts::UnitConstants::mm;
```
Comment (paulgessinger) [inline Core/include/Acts/Seeding/SeedfinderConfig.hpp:92]: @robertlangenberg Are these in `mm`?

File: Core/include/Acts/Seeding/SeedFilterConfig.hpp (hunk 1 above)
Comment (robertlangenberg) [reply Core/include/Acts/Seeding/SeedFilterConfig.hpp:21]: Yes

File: Core/include/Acts/Seeding/SeedFilterConfig.hpp [hunk 4]
```diff
@@ -16,21 +18,30 @@ namespace Acts {
...
   unsigned int maxSeedsPerSpM = 10;
   // how often do you want to increase the weight of a seed for finding a
   // compatible seed?
   size_t compatSeedLimit = 2;
   // Tool to apply experiment specific cuts on collected middle space points
+
+  SeedFilterConfig toInternalUnits() {
+    using namespace Acts::UnitLiterals;
+    SeedFilterConfig config = *this;
+    config.deltaRMin /= 1_mm;
```
Comment (robertlangenberg) [inline Core/include/Acts/Seeding/SeedFilterConfig.hpp:40]: that's just mm, not inverse mm

File: Core/include/Acts/Seeding/SeedFilterConfig.hpp [hunk 5]
```diff
@@ -16,21 +18,30 @@ namespace Acts {
...
+    config.deltaInvHelixDiameter /= 1. / 1_mm;
```
Comment (robertlangenberg) [inline Core/include/Acts/Seeding/SeedFilterConfig.hpp:41]: am I reading this wrong? invHelixDiam/1/mm = invHelixDiam*mm?

File: Core/include/Acts/Seeding/SeedfinderConfig.hpp [hunk 6]
```diff
@@ -23,16 +24,15 @@ struct SeedfinderConfig {
...
+  // lower cutoff for seeds
+  float minPt = 400. * Acts::UnitConstants::MeV;
   // cot of maximum theta angle
   // equivalent to 2.7 eta (pseudorapidity)
   float cotThetaMax = 7.40627;
-  // minimum distance in mm in r between two measurements within one seed
-  float deltaRMin = 5;
-  // maximum distance in mm in r between two measurements within one seed
-  float deltaRMax = 270;
+  // minimum distancein r between two measurements within one seed
```
Comment (robertlangenberg) [inline Core/include/Acts/Seeding/SeedfinderConfig.hpp:32]:
```suggestion
  // minimum distance in r between two measurements within one seed
```

File: Core/include/Acts/Seeding/SeedfinderConfig.hpp (hunk 2 above)
Comment (robertlangenberg) [reply Core/include/Acts/Seeding/SeedfinderConfig.hpp:51]: correct

File: Core/include/Acts/Seeding/SeedFilterConfig.hpp (hunk 5 above)
Comment (robertlangenberg) [reply Core/include/Acts/Seeding/SeedFilterConfig.hpp:41]: my bad

Comment (paulgessinger) [reply Core/include/Acts/Seeding/SeedFilterConfig.hpp:41]: Yes, exactly. I'm only writing it this way so I'm dividing consistently.

File: Examples/Run/Reconstruction/Common/SeedingExample.cpp [hunk 7]
```diff
@@ -121,21 +121,21 @@ int runSeedingExample(int argc, char* argv[],
...
-  seedingCfg.zMax = 2000.;
+  seedingCfg.rMax = 100._mm;
+  seedingCfg.deltaRMax = 60._mm;
+  seedingCfg.collisionRegionMin = -250_mm;
+  seedingCfg.collisionRegionMax = 250._mm;
+  seedingCfg.zMin = -2000._mm;
+  seedingCfg.zMax = 2000._mm;
   seedingCfg.maxSeedsPerSpM = 1;
   seedingCfg.cotThetaMax = 7.40627;  // 2.7 eta
   seedingCfg.sigmaScattering = 50;
```
Comment (robertlangenberg) [inline Examples/Run/Reconstruction/Common/SeedingExample.cpp:132]:
```suggestion
  seedingCfg.sigmaScattering = 5;
```

I know it's not really related...

**Review by robertlangenberg — APPROVED** [review]
Thanks a bunch for this!

File: Examples/Run/Reconstruction/Common/SeedingExample.cpp (hunk 7 above)
Comment (paulgessinger) [reply Examples/Run/Reconstruction/Common/SeedingExample.cpp:132]: I think I'd prefer doing this in it's own PR to decouple it from the purely technical changes here.

**Comment by asalzburger** [comment]
This needs to go in before #945 I understand ... ok le't wait for that then.

**Pull request merged**